akka-actor/src/main/scala/akka/io/Tcp.scala

```diff
@@ -431,7 +431,23 @@ object Tcp extends ExtensionId[TcpExt] with ExtensionIdProvider {
    * Whenever a command cannot be completed, the queried actor will reply with
    * this message, wrapping the original command which failed.
    */
-  final case class CommandFailed(cmd: Command) extends Event
+  final case class CommandFailed(cmd: Command) extends Event {
+    private var _cause: Option[Throwable] = None
```

Comment:
Do we guarantee that events are serializable? We might then want to mark this field `@transient` just to be sure.